src/runtime/ValueInlines.h

```diff
@@ -607,6 +607,20 @@ ALWAYS_INLINE bool Value::isInt32ConvertibleDouble(const double& d, int32_t& asI
     return true;
 }
 
+inline Value::Value(NaNInitTag)
+{
+    *this = Value(EncodeAsDouble, std::numeric_limits<double>::quiet_NaN());
+}
+
+inline Value::Value(InfinityInitTag tag)
+{
+    if (tag == InfinityInitTag::PostiveInfinityInit) {
+        *this = Value(EncodeAsDouble, std::numeric_limits<double>::infinity());
+    } else {
+        *this = Value(EncodeAsDouble, -std::numeric_limits<double>::infinity());
+    }
+}
+
 inline Value::Value(const double& d)
 {
     int32_t asInt32;
```

Comment:
It seems that `Value::Value(const double& d)` method could not process Nan or infinite value correctly for arm-clang environment. 
I wonder what the main reason of the bug was

Reply:
arm-clang could not process well when input ctor to const inf, nan value.
I don't know why if could happen.
If we use O0~1 options, it is not happend

Reply:
I see. Thanks for your great work 👏